app/src/main/java/org/projectbuendia/client/ui/OdkActivityLauncher.java

```diff
@@ -102,6 +132,11 @@ public static void fetchAndShowXform(
         final int requestCode,
         @Nullable final org.odk.collect.android.model.Patient patient,
         @Nullable final Preset fields) {
+        LOG.i("Trying to fetch it from cache.");
```

Comment:
This appears to be the other way around from how you describe in the PR description.

It looks like it tries the cache first. If it succeeds in getting it from the local cache, it returns. If it doesn't, it tries to fetch it from the server.

That does sound like it will be faster as it doesn't have to go to the server each time.

The implication is that if a form is changed, the new version will never be retrieved unless that cache is invalidated. Therefore we probably need some way to clear the forms cache and force it to get the forms from the server (maybe in the settings).